homeassistant/components/surepetcare/sensor.py

```diff
@@ -0,0 +1,158 @@
+"""Support for Sure PetCare Flaps binary sensors."""
+import logging
+import pprint
+
+import homeassistant.helpers.device_registry as dr
+from homeassistant.const import (ATTR_VOLTAGE, CONF_ID, CONF_NAME, CONF_TYPE,
+                                 DEVICE_CLASS_BATTERY)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import (BATTERY_ICON, CONF_HOUSEHOLD_ID, DATA_SURE_PETCARE,
+                    DATA_SUREPY, SURE_BATT_VOLTAGE_DIFF, SURE_BATT_VOLTAGE_LOW,
+                    SURE_IDS, TOPIC_UPDATE, SureProductID, SureThingID)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up Sure PetCare Flaps sensors based on a config entry."""
+    from surepy import SurePetcare
+
+    entities = list()
+    surepy: SurePetcare = hass.data[DATA_SURE_PETCARE][DATA_SUREPY]
+
+    for thing in hass.data[DATA_SURE_PETCARE][SURE_IDS]:
+        sure_id = thing[CONF_ID]
+        sure_type = thing[CONF_TYPE]
+
+        if sure_type != SureThingID.FLAP.name:
+            continue
+
+        if sure_id not in hass.data[DATA_SURE_PETCARE][sure_type]:
+            hass.data[DATA_SURE_PETCARE][
+                sure_type][sure_id] = await surepy.get_flap_data(sure_id)
+
+        entities.append(FlapBattery(sure_id, thing[CONF_NAME], hass=hass))
+
+    async_add_entities(entities, True)
+
+
+class FlapBattery(Entity):
+    """Sure Petcare Flap."""
+
+    def __init__(self, _id: int, name: str, data: dict = None, hass=None):
+        """Initialize a Sure Petcare Flap battery sensor."""
+        self._id = _id
+        self._hass = hass
+        self._name = f"Flap {name.capitalize()} Battery Level"
+        self._unit_of_measurement = "%"
+        self._icon = BATTERY_ICON
+        self._device_class = DEVICE_CLASS_BATTERY
```

Comment:
We should implement the `device_class` property using the `DEVICE_CLASS_BATTERY` constant.